3-networks/modules/vpn-ha/main.tf

```diff
@@ -0,0 +1,199 @@
+  network    = var.network_name
+  name       = "vpn-${var.vpc_name}-${var.default_region1}-cr1"
+  peer_external_gateway = {
+    redundancy_type = "SINGLE_IP_INTERNALLY_REDUNDANT"
+    interfaces = [{
+      id         = 0
+      ip_address = var.on_prem_ip_address
+
+    }]
+  }
+  router_name = var.region1_router1_name
+  tunnels = {
+    remote-0 = {
+      bgp_peer = {
+        address = var.remote0_ip
+        asn     = var.bgp_asn
+      }
+      bgp_peer_options                = null
+      bgp_session_range               = var.remote0_range
+      ike_version                     = 2
+      vpn_gateway_interface           = 0
+      peer_external_gateway_interface = 0
+      shared_secret                   = var.remote0_secret
+    }
+    remote-1 = {
+      bgp_peer = {
+        address = var.remote1_ip
+        asn     = var.bgp_asn
+      }
+      bgp_peer_options                = null
+      bgp_session_range               = var.remote1_range
+      ike_version                     = 2
+      vpn_gateway_interface           = 1
+      peer_external_gateway_interface = 0
+      shared_secret                   = var.remote1_secret
+    }
+  }
+}
+
+module "vpn_ha_region1_router2" {
+  source     = "terraform-google-modules/vpn/google//modules/vpn_ha"
+  project_id = local.interconnect_project_id
+  region     = var.default_region1
+  network    = var.network_name
+  name       = "vpn-${var.vpc_name}-${var.default_region1}-cr2"
+  peer_external_gateway = {
+    redundancy_type = "SINGLE_IP_INTERNALLY_REDUNDANT"
+    interfaces = [{
+      id         = 0
+      ip_address = var.on_prem_ip_address
+
+    }]
+  }
+  router_name = var.region1_router2_name
+  tunnels = {
+    remote-0 = {
+      bgp_peer = {
+        address = var.remote0_ip
+        asn     = var.bgp_asn
+      }
+      bgp_peer_options                = null
+      bgp_session_range               = var.remote0_range
+      ike_version                     = 2
+      vpn_gateway_interface           = 0
+      peer_external_gateway_interface = 0
+      shared_secret                   = var.remote0_secret
+    }
+    remote-1 = {
+      bgp_peer = {
+        address = var.remote1_ip
+        asn     = var.bgp_asn
+      }
+      bgp_peer_options                = null
+      bgp_session_range               = var.remote1_range
+      ike_version                     = 2
+      vpn_gateway_interface           = 1
+      peer_external_gateway_interface = 0
+      shared_secret                   = var.remote1_secret
+    }
+  }
+}
+
+module "vpn_ha_region2_router1" {
+  source     = "terraform-google-modules/vpn/google//modules/vpn_ha"
+  project_id = local.interconnect_project_id
+  region     = var.default_region2
+  network    = var.network_name
+  name       = "vpn-${var.vpc_name}-${var.default_region2}-cr1"
+  peer_external_gateway = {
+    redundancy_type = "SINGLE_IP_INTERNALLY_REDUNDANT"
+    interfaces = [{
+      id         = 0
+      ip_address = var.on_prem_ip_address
```

Comment:
Across two regions, we would probably expect to have two different on-prem IPs rather than one right?

Reply:
Yes, already fixed :)